src/settings.rs

```diff
@@ -16,18 +16,18 @@ pub const APP_TOOL_TIP: &str = "<FILE> [OPTIONS]";
 
 /// This application's version information.
 ///
-/// This constant represents this application's major version.
+/// This constant represents this application's fix level.
 #[allow(dead_code)]
-pub const VERSION_MAJOR: u8 = 0u8;
+pub const VERSIO_FIX_LEVEL: u8 = 0u8;
 
 /// This application's version information.
 ///
-/// This constant represents this application's minor version.
+/// This constant represents this application's major version.
 #[allow(dead_code)]
-pub const VERSION_MINOR: u8 = 1u8;
+pub const VERSION_MAJOR: u8 = 0u8;
 
 /// This application's version information.
 ///
-/// This constant represents this application's fix level.
+/// This constant represents this application's minor version.
 #[allow(dead_code)]
-pub const VERSIO_FIX_LEVEL: u8 = 0u8;
+pub const VERSION_MINOR: u8 = 1u8;
```

Comment:
I prefer symbols to be ordered by alphabet if their order has no affect to the overall behaviour.  But when I introduced the version constants, I forgot to check for their order before I loaded up the respective commits.  When the change was already on GitHub, I needed to fix the order with a new commit.

Reply:
I think it would be more logical to have them in a major, minor, fix order, since these numbers have a clear order. But as noted in my other comment, I think removing these makes generally more sense.

You can use `git rebase` to change the order of your commits, edit their commit messages or also combine multiple into one in different ways. I normally use it always local, but I've definitely seen it done in other pull requests.

Reply:
Throwing away a backup of the repository due to the history having changed is quite easy.  But throwing away the GitHub repository in case it should have become incompatible with the local repository might be more difficult.  How to handle this case best?

Reply:
I don't know if I get your message.. What do you mean?

Reply:
When pushing to another repository, your history does need to incorporate the history of the other repository.  You can test what would happen if you do a `git clone --bare` of your repository, move the copy to another directory and set it up as push destination.  I will call this the `backupdestination`, in the following.

Now rewrite the history of your repository and push to `backupdestination`.  Git will deny this since you need to incorporate the history of the `backupdestination` before the push.  But this is what you intended to change by rewriting the history.  The simplest option is to throw away the `backupdestination` and do a new `git clone --bare` before setting up the push destination again.

Doing so with your GitHub repository will break even more -- break in the meaning of "to shatter".  This is why I consider editing already pushed commits a problem and try to avoid it.

These complications occur when amending commits, squashing branches or editing commit messages.  Once pushed, you are done and changes are impossible.

Now:  how to edit commit messages from long ago without breaking the whole GitHub repository?  (break in the meaning of "to shatter".)

Reply:
Just quickly looking it up, it seems simple force pushing is accepted in that case:

[https://stackoverflow.com/questions/63568107/changing-commit-message-in-github-pull-request](https://stackoverflow.com/questions/63568107/changing-commit-message-in-github-pull-request)

Reply:
:wrench: :heavy_check_mark: